src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -298,38 +300,30 @@
 
   The cursor is closed when `func` returns. If an exception is thrown,
   the query is cancelled."
-  [func sql params]
+  [[sql & params] func]
```

Comment:
I see how it reduces duplication, but this is a very strange function signature.